src/response.ts

```diff
@@ -1,45 +1,64 @@
+import { debug } from 'console';
 import { Millisecond } from 'italia-ts-commons/lib/units';
 import { DeferredPromise } from 'italia-ts-commons/lib/promises';
+import { fromNullable, none } from 'fp-ts/lib/Option';
+import { toError } from 'fp-ts/lib/Either';
+import { fromPredicate } from 'fp-ts/lib/TaskEither';
 import { Client, createClient } from '../generated/definitions/pagopa/client';
 import { TransactionStatusResponse } from '../generated/definitions/pagopa/TransactionStatusResponse';
+import { getUrlParameter } from './js/urlUtilities';
 import idpayguard from './js/idpayguard';
 import { initHeader } from './js/header';
 import { setTranslateBtns } from './js/translateui';
 import { initDropdowns } from './js/dropdowns';
 import { constantPollingWithPromisePredicateFetch, retryingFetch } from './utils/fetch';
 import {
+  getTransactionFromSessionStorageTask,
+  getStringFromSessionStorageTask,
+  resumeTransactionTask,
   checkStatusTask,
-  getDataFromSessionStorageTask,
-  isNot3dsFlowTask,
-  showErrorStatus,
-  showSuccessStatus,
 } from './utils/transactionHelper';
+import { start3DS2MethodStep, createIFrame, start3DS2AcsChallengeStep } from './utils/iframe';
 
 // eslint-disable-next-line sonarjs/cognitive-complexity
 document.addEventListener('DOMContentLoaded', async () => {
   const retries: number = 10;
   const delay: number = 3000;
   const timeout: Millisecond = 20000 as Millisecond;
 
-  const isTransientErrorGivenFinalStatus = async (response: Response): Promise<boolean> => {
-    const transactionStatus = TransactionStatusResponse.encode(await response.clone().json());
-    return response.status === 200 && transactionStatus.data?.finalStatus === false;
-  };
+  const paymentManagerClientWithPollingOnMethod: Client = createClient({
+    baseUrl: 'http://localhost:8080',
+    fetchApi: constantPollingWithPromisePredicateFetch(
+      DeferredPromise<boolean>().e1,
+      retries,
+      delay,
+      timeout,
+      async (r: Response): Promise<boolean> => {
+        const myJson = (await r.clone().json()) as TransactionStatusResponse;
+        // Stop the polling when this condition is false
+        return fromNullable(myJson.data.methodUrl).isNone();
```

Comment:
After this change I'think we can said about _TransactionStatesTypes_  _"give it to the cat" _ ( cit A. Sordi) , or not ?